Shell/FunctionDefinitionHandler.cpp

```diff
@@ -0,0 +1,495 @@
+/*
+ * This file is part of the source code of the software program
+ * Vampire. It is protected by applicable
+ * copyright laws.
+ *
+ * This source code is distributed under the licence found here
+ * https://vprover.github.io/license.html
+ * and in the source directory
+ */
+
+#include "FunctionDefinitionHandler.hpp"
+#include "Inferences/InductionHelper.hpp"
+
+#include "Kernel/TermIterators.hpp"
+#include "Kernel/Signature.hpp"
+#include "Kernel/Problem.hpp"
+
+#include "Lib/Hash.hpp"
+#include "Lib/SharedSet.hpp"
+
+using namespace Inferences;
+using namespace Kernel;
+using namespace Lib;
+using namespace std;
+
+namespace Shell {
+
+inline bool canBeUsedForRewriting(Term* lhs, Clause* cl)
```

Comment:
`inline`?

Reply:
You mean inlining is not needed in this case?

Reply:
I mean that I don't think it's required for the linkage semantics. For hinting to the compiler that the function should be inlined (in the same translation unit), no objection, but generally I'd let the compiler decide unless you have good evidence one way or another.

Reply:
Just wondering; no need to change it.

Reply:
Makes sense, I removed the `inline`.